pyang/translators/yang.py

```diff
@@ -366,7 +366,7 @@ def select_quote(s, q, pref_q):
             # the string was double quoted for a reason, keep it
             return '"'
     elif q == "'":
-        if need_quote(s):
+        if need_quote(s, skip_space=True):
```

Comment:
I think that maybe it was a mistake to re-use the function need_quote() in this case.  need_quote() is used (originally) to decide ifa string can be unquoted, or needs quotes.  In this case, we want to check if the string needs single-quote, or can use the preferred double-quote.  So I think we should leave need_quote() as it was, and introduce a new function need_single_quote() here.